### PR TITLE
Added sentry

### PR DESCRIPTION
**Pull request by suyashdeshpande**
For issue  #19 
@hereisnaman pls review

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -5,27 +5,57 @@ const withCss = require('@zeit/next-css');
 const withSass = require('@zeit/next-sass');
 const withFonts = require('next-fonts');
 
+const SentryWebpackPlugin = require('@sentry/webpack-plugin');
```
Comment (0xnmn) [inline next.config.js]: Do not need the release integration. Remove the Webpack Plugin.

File: app/pages/_app.js [hunk 2]
```diff
@@ -13,6 +14,11 @@ import 'Styles/theme/styles.scss';
 import 'Styles/theme/iconsmind/style.scss';
 import 'Styles/theme/simple-line-icons/css/simple-line-icons.css';
 
+// Todo: Add SENTRY_DSN in your .env file
```
Comment (0xnmn) [inline app/pages/_app.js]: Remove the comment.

File: app/pages/_app.js [hunk 3]
```diff
@@ -13,6 +14,11 @@ import 'Styles/theme/styles.scss';
...
+Sentry.init({
```
Comment (0xnmn) [inline app/pages/_app.js:17]: Add custom err handle in App.componentDidCatch

**Review by 0xnmn — CHANGES_REQUESTED** [review]

**Comment by 0xnmn** [comment]
@suyashdeshpande No sentry integration has been made api side. Also need to have two separate sentry namespaces for api and client.

**Comment by suyashdeshpande** [comment]
@hereisnaman in webpack config, I have replaced @sentry/node with  @sentry/browser if it runs at the client-side. Won't it work for both? And what do you mean by separate sentry namespaces for API and client? Can you help ?

**Comment by 0xnmn** [comment]
@suyashdeshpande no it won't work for the api and we don't need the webpack integration.

By separate namespace I mean, there will be separate DSN for the client and the API so you'll have to use and name the env vars accordingly.

File: next.config.js (hunk 1 above)
Comment (suyashdeshpande) [reply next.config.js]: Done

File: app/pages/_app.js (hunk 2 above)
Comment (suyashdeshpande) [reply app/pages/_app.js]: Done

**Comment by suyashdeshpande** [comment]
@hereisnaman can you please review. I have made the necessary changes.

**Comment by 0xnmn** [comment]
There is still no sentry integration for the graphql api

**Comment by suyashdeshpande** [comment]
@hereisnaman in which file do I need to integrate sentry for graphql as there are multiple files with initialization of ApolloServer ?
1.  api/routes/graphql/index.js
2.  app/pages/api/graphql/index.js
3.  pages/api/graphql/index.js
Can you help ?

**Comment by 0xnmn** [comment]
> @hereisnaman in which file do I need to integrate sentry for graphql as there are multiple files with initialization of ApolloServer ?
> 
>     1. api/routes/graphql/index.js
> 
>     2. app/pages/api/graphql/index.js
> 
>     3. pages/api/graphql/index.js
>        Can you help ?

Use the [formatError](https://www.apollographql.com/docs/apollo-server/data/errors/#for-the-client-response) function to track the graphql errors. 

Also integrate [here](https://github.com/coding-blocks/gondor/blob/master/api/controllers/Base.js#L7)

**Comment by 0xnmn** [comment]
Closing due to inactivity.

**Pull request closed without merging**